### PR TITLE
fix: add GHE support, and made custom hook to handle requests

### PR DESCRIPTION
**Pull request by lewtakm**
This PR adds support for GHE users and add custom hook to handle request since there is more and more code that was repeated by each card.

### CODE REVIEW

**Review by mcalus3 — APPROVED** [review]

File: src/components/useRequest.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  > => {
+    const token = await auth.getAccessToken(['repo']);
+    const octokit = new Octokit({auth: token});
+    const response = await octokit.request(`GET /repos/{owner}/{repo}/${requestName}`, {
+      ...(baseUrl && {baseUrl}),
+      owner,
+      repo,
+      ...(perPage && {per_page: perPage}),
+    });
+    const data = await response.data;
```
Comment (dtuite) [inline src/components/useRequest.ts:47]: What is happening in this line? Is this accessing the `data` property of the response? How can that be an async operation. I would have expected it to be `const data = response.data` or `const data = await response.data()`.

Comment (dtuite) [reply src/components/useRequest.ts:47]: Perhaps there is just something about Typescript that I don't understand.

**Comment by dtuite** [comment]
@lewtakm should there be a version bump included in this commit?

**Comment by adamdmharvey** [comment]
This is awesome; thanks!  One thing i was thinking about also was the annotation.  Today it expects `github.com/project-slug`, which of course is not theoretically valid for a GHE instance.

Almost think the annotation would need to be changed to `github/project-slug`?

Longer term, that might causes confusion if a Backstage instance has both a GHE and a GitHub.com connection. (e.g., where my firm has most of our code inner source, but we do contribute with open source and may want to refer to our open source projects in future, and we may want one project to have a slug for org/repo on GHE and another project have an annotation for org/repo on GitHub.com).  Though not having this isn't a short term issue I don't believe, but good for strategy?

**Review by adamdmharvey — COMMENTED** [review]
Wonder if we need this updated as well?

https://github.com/RoadieHQ/backstage-plugin-github-insights/blob/main/src/components/Widgets/ContributorsCard/components/Contributor/Contributor.tsx#L43

File: src/components/useRequest.ts (hunk 1 above)
Comment (lewtakm) [reply src/components/useRequest.ts]: Octokit.request return Promise but you are right second await is not needed here.

**Pull request merged**